### PR TITLE
CDRIVER-4362 Add showExpandedEvents option for C2C replication

### PR DESCRIPTION
**Pull request by galon1**
Support for the `showExpandedEvents` option is added, following the spec changes in this [ticket](https://jira.mongodb.org/browse/DRIVERS-1975). A new test was added following this [PR](https://github.com/mongodb/specifications/commit/9652564e6576fc541c5a984caafea2188e86abb4). No new changes were needed for the new `changeStream` events, but the `dropIndex` which was added to the test runner.

Note: In order for the tests to pass this PR will also merge the fix listed in [CDRIVER-4464](https://jira.mongodb.org/browse/CDRIVER-4464)

### CODE REVIEW

File: src/libmongoc/tests/unified/operation.c [hunk 1]
```diff
@@ -2154,6 +2154,50 @@ operation_close (test_t *test,
...
+   bool ret = false;
+   entity_t *entity;
+   bson_parser_t *parser = NULL;
+   char *index = NULL;
+   bson_t *opts = NULL;
+   mongoc_collection_t *coll = NULL;
+   bson_error_t op_error = {0};
+
+   parser = bson_parser_new ();
+   bson_parser_allow_extra (parser, true);
```
Comment (kevinAlbs) [inline src/libmongoc/tests/unified/operation.c:2172]:
```suggestion
```

IMO it is preferable not to allow extra fields in operations parsed from the specification test files. Unless those extra fields are used as an input for something else. Disallowing extra fields will immediately inform us if a spec test file adds an unexpected argument.

**Review by vector-of-bool — APPROVED** [review]
LGTM

File: src/libmongoc/src/mongoc/mongoc-opts-private.h [hunk 2]
```diff
@@ -148,6 +148,7 @@ typedef struct _mongoc_change_stream_opts_t {
    const char *fullDocumentBeforeChange;
    bson_value_t comment;
    bson_t extra;
+   bool showExpandedEvents;
```
[the remark this answers is not in the document]
Comment (kevinAlbs) [reply src/libmongoc/src/mongoc/mongoc-opts-private.h:151]: Also run `python ./build/generate-opts.py` to regenerate the files. That should update the [option documentation](http://mongoc.org/libmongoc/current/mongoc_collection_watch.html#parameters).

**Review by kevinAlbs — APPROVED** [review]
LGTM with running `python ./build/generate-opts.py` to regenerate the option documentation.

**Pull request merged**